conformance/tests/httproute-disallowed-kind.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: Gateway
+metadata:
+  name: tlsroutes-only
+  namespace: gateway-conformance-infra
+spec:
+  gatewayClassName: {GATEWAY_CLASS_NAME}
+  listeners:
+  - name: tls
+    port: 443
+    protocol: TLS
+    tls:
+      mode: Passthrough
+    allowedRoutes:
+      namespaces:
+        from: Same
+      kinds:
+      - kind: TLSRoute
```

Comment:
Per https://github.com/kubernetes-sigs/gateway-api/issues/1104#issuecomment-1112515816 it may be problematic for some implementations not to have the Gateway definitions in the base manifests, need to sort out the pattern here.

Reply:
cc @hbagdi and @shaneutt

Reply:
For reference, the alternative here would be to add more Gateways to our base set of resources instead. That would work, but could become difficult to maintain.

Reply:
Discussed in community meeting on Monday in reference to this and https://github.com/kubernetes-sigs/gateway-api/pull/1137#discussion_r863243598. Apparently this ended up being a non issue for implementations now.